crates/sui-framework/packages/deepbook/sources/clob.move

```diff
@@ -917,16 +936,19 @@ module deepbook::clob {
             return (base_quantity_filled, quote_quantity_filled, true, order_id)
         } else {
             assert!(restriction == NO_RESTRICTION, EInvalidRestriction);
-            order_id = inject_limit_order(
-                pool,
-                price,
-                quantity - base_quantity_filled,
-                is_bid,
-                expire_timestamp,
-                account_cap,
-                ctx
-            );
-            return (base_quantity_filled, quote_quantity_filled, true, order_id)
+            if(quantity - base_quantity_filled > 0){
+                order_id = inject_limit_order(
+                    pool,
+                    price,
+                    quantity - base_quantity_filled,
+                    is_bid,
+                    expire_timestamp,
+                    account_cap,
+                    ctx
+                );
+                return (base_quantity_filled, quote_quantity_filled, true, order_id)
+            };
+            return (base_quantity_filled, quote_quantity_filled, false, 0)
```

Comment:
- Please add a comment explaining the `order_id = 0` return value + what it means
- Please add a test that exercises the 0 return value

Reply:
we have add extra code comments to explain the return values.

Reply:
Could you add these as doc comments (`///`) on the function itself? It will be difficult for folks to discover these as internal comments.